### PR TITLE
feat: Ruff and Poetry Github Actions (#2)

### LINKED ISSUE #2: [Feat]: Create GitHub actions
### My suggestion

Add new [GitHub action](https://docs.github.com/en/actions/using-workflows/about-workflows) to `.github/workflows` It should contain things like:
 - `ruff` code checks
 - `poetry check`

### Possible alternatives

_No response_

### PR DESCRIPTION
**Pull request by Sanielia**
- Added YML that performs ruff lints and poetry checks on push

> [!IMPORTANT]
> It's my first time doing github actions, I have used LLMs and Google, but still it is probable, that I have made some huge mistakes

### CODE REVIEW

File: .github/workflows/ruff-code-checks-and-poetry-cecks.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Ruff code checks and poetry checks
+on: push
+
+jobs:
+  ruff-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - uses: chartboost/ruff-action@v1
```
Comment (Norbiros) [inline .github/workflows/ruff-code-checks-and-poetry-cecks.yml]: I don't like systems where it auto commits some changes. We already have pre-commit hook where it runs `ruff fix`.
This gh action should detect if code is not formatted/has errors and then it should raise error.

File: .github/workflows/ruff-code-checks-and-poetry-cecks.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: 'style fixes by ruff'
+  poetry-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - run:
+          poetry update
```
Comment (Norbiros) [inline .github/workflows/ruff-code-checks-and-poetry-cecks.yml:29]: Poetry isn't installed by default. Maybe use https://github.com/marketplace/actions/python-poetry-action

**Review by Norbiros — CHANGES_REQUESTED** [review]

File: .github/workflows/ruff-code-checks-and-poetry-cecks.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+name: Ruff code checks and poetry checks
```
Comment (Norbiros) [inline .github/workflows/ruff-code-checks-and-poetry-cecks.yml]: You could rename it to something like `Code quality checks` or something less specific imo...

**Review by Norbiros — COMMENTED** [review]
Almost perfect!

**Review by Norbiros — APPROVED** [review]

**Review by goteusz-maszyk — APPROVED** [review]

**Pull request merged**